src/chart/Scatter.js

```diff
@@ -0,0 +1,171 @@
+                var point = element.selectAll(".point").data(d.filter(function(d, i) {return i > 0;}));
+                var point1 = element.selectAll(".point").data(d.filter(function(d, i) {return i > 0;}));
+                switch (context.dotRepresentation()) {
+                    case "circle" :
+                        var title = point
+                          .enter().append("circle")
+                            .attr("class", "point")
+                            .attr("r", context.pointSize()/2)
+                            .on("click", function (d2, idx) {
+                                context.click(context.rowToObj(d), context._columns[idx + 1]);
+                            })
+                            .append("title")
+                        ;
+                        break;
+
+                    case "rectangle" :
+                        var title = point
+                          .enter().append("rect")
+                            .attr("class", "point")
+                            .attr("width", context.pointSize())
+                            .attr("height", context.pointSize())
+                            .on("click", function (d2) {
+                                context.click(context.rowToObj(d), context._columns[idx + 1]);
+                            })
+                            .append("title")
+                        ;
+                        break;
+
+                    case "cross" :
+                        var title = point
+                          .enter().append("line")
+                            .attr("class", "point")
+                            .attr("stroke-width", 2)
+                            .on("click", function (d2) {
+                                context.click(context.rowToObj(d), context._columns[idx + 1]);
+                            })
+                            .append("title")
+                        ;
+                        var title1 = point1
+                            .enter().append("line")
+                            .attr("class", "point")
+                            .attr("stroke-width", 2)
+                            .on("click", function (d2) {
+                                context.click(context.rowToObj(d), context._columns[idx + 1]);
+                            })
+                            .append("title1")
+                        break;
+                }
+
+                if (context.orientation() === "horizontal") {
+                    switch (context.dotRepresentation()) {
+                        case "circle" :
+                            point.transition()
+                                .attr("cx", function (d2, idx) { return context.dataScale(d[0]) + context.dataScale.rangeBand()/2 ;})
+                                .attr("cy", function (d2, idx) { return context.valueScale(d2); })
+                                .style("fill", function (d2, idx) { return context._palette(idx + 1); })
```

Comment:
All calls to "_palette" should use the column label and not the idx so that the colours are consistent with other charts displaying the same data...